<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Juozas Gordevičius" />


<title>Cancer related methylation effects in blood</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cancer Effects in Blood</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="GSE19711.html">GSE19711</a>
</li>
<li>
  <a href="GSE30229.html">GSE30229</a>
</li>
<li>
  <a href="GSE50409.html">GSE50409</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Cancer related methylation effects in blood</h1>
<h4 class="author"><em>Juozas Gordevičius</em></h4>

</div>


<div id="the-background" class="section level1">
<h1>The background</h1>
<p>Now that we know some basic R and design of basic experiments, we can try to answer simple questions.</p>
<p>We will use publically available data to answer a compelling question:</p>
<ul>
<li>Is there a difference in blood DNA modification between cancer patients and healthy individuals?</li>
<li>Let’s look into three different datasets and try to find a pattern.</li>
</ul>
</div>
<div id="the-data" class="section level1">
<h1>The data</h1>
<p>Three different datasets pertaining to different types of cancer are available</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE50409">GSE50409</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE30229">GSE30229</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19711">GSE19711</a></li>
</ul>
<div id="the-platform" class="section level2">
<h2>The platform</h2>
<p>Illumina 27k is microarrays have ~ 27.000 probes. Each of the probes interrogates a CG dinucleotide in the genome. Each probe yields two signals: the methylation intensity (M) and unmethylation intensity (U). The final modification score is defined as</p>
<p><span class="math display">\[
\beta = \frac{M}{M+U}
\]</span></p>
<p>Read this paper for better understanding:</p>
<ul>
<li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-587">Beta vs M-value</a></li>
</ul>
</div>
</div>
<div id="the-workflow" class="section level1">
<h1>The workflow</h1>
<ul>
<li>Proceed in small steps</li>
<li>Read online as much as you can</li>
<li>Send pull requests (PRs) often</li>
<li>I will comment and suggest further directions</li>
</ul>
<div id="step-1---getting-started" class="section level2">
<h2>Step 1 - getting started</h2>
<ul>
<li>Choose your group mates</li>
<li>Choose the dataset you are going to analyse</li>
<li>Fork this repo</li>
<li>Enter your name in corresponding Rmd file</li>
<li>You can create a slack channel for discussions within your group</li>
<li>Send PR so that I see how the groups develop</li>
</ul>
</div>
<div id="step-2" class="section level2">
<h2>Step 2</h2>
<p>Minfi is the main tool for processing Illumina methylation arrays. Read the guides!</p>
<ul>
<li><a href="https://www.bioconductor.org/help/course-materials/2014/BioC2014/minfi_BioC2014.pdf">Minfi tutorial</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/vignettes/minfi/inst/doc/minfi.pdf">The minfi user’s guide</a></li>
</ul>
<p>GEOQuery is a good tool to download data from GEO. But may not be the best solution. A <code>file.download</code> command may be a good alternative.</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html">GEOQuery</a></li>
</ul>
<p>Write the code that prepares the data:</p>
<ul>
<li><p>Automatically download the data from GEO</p></li>
<li><p>Obtain the matrix of beta values where each row corresponds to probes and each column corresponds to samples</p></li>
<li><p>How many samples and how many probes do you have in the data?</p></li>
<li><p>How are the beta values distributed?</p></li>
<li><p>Do your probes have names?</p></li>
<li><p>Do you have annotation that tells the coordinate (in hg19) of each probe and its genomic features (such as related gene name)?</p></li>
<li><p>Do you know which samples correspond to healthy individuals, and which samples correspond to the sick ones?</p></li>
<li><p>Send PR!</p></li>
</ul>
</div>
<div id="step-3" class="section level2">
<h2>Step 3</h2>
<p>Start with the most basic approach</p>
<ul>
<li><p>For each probe compute a t-test to verify if the distributions of beta values within the probe significantly differ between the two groups.</p></li>
<li><p>From the t-test, obtain the p value.</p></li>
<li><p>Plot the distribution of p values. What is the expected distribution? How dows it differ from what you get?</p></li>
<li><p>Performance-wise, how long will it take to compute the test for all probes?</p></li>
<li><p>Can you suggest a faster alternative for computation?</p></li>
<li><p>Send PR!</p></li>
</ul>
</div>
<div id="step-4" class="section level2">
<h2>Step 4</h2>
<p>Let’s deal with the p values.</p>
<ul>
<li><p>What is multiple hypothesis testing?</p></li>
<li><p>How should we adjust for multiple hypothesis testing in our case?</p></li>
<li><p>Did you find any probes that show statistically significant modification difference between healthy and sick individuals?</p></li>
<li><p>Where are these probes? What genes are they related to?</p></li>
<li><p>Send PR!</p></li>
</ul>
</div>
<div id="next-steps" class="section level2">
<h2>Next steps</h2>
<p>Once we get here, we can work on the other issues:</p>
<ul>
<li><p>Normalizavimas, kvantiliu normalizacija</p></li>
<li><p>PCA</p></li>
<li><p>Reikia atmesti lytiniu chromosomu probus</p></li>
<li><p>Some samples could be outliers messing up our results. How to detect outliers?</p></li>
<li><p>Thousands of t-tests can be computed at once within seconds. How? limma</p></li>
<li><p>Permutacijos!!</p></li>
<li><p>Remember that the experiment may have more than one variable (disease). Age may be a factor two. How can we account for differences in age?</p></li>
<li><p>Age and Blood cellular composition may be an issue too. Maybe the differences we observe between patient groups are not actual methylation differences but artefacts of varying blood cellular composition? How shall we deal with that?</p></li>
<li><p>Rezultatas: pakite probai!!!</p></li>
<li><p>Once we have analysis of all three datasets, can we pull them all together and make conclusions?</p></li>
</ul>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
